src/mpid/ch4/netmod/ucx/ucx_recv.h

```diff
@@ -24,10 +24,13 @@ MPL_STATIC_INLINE_PREFIX void MPIDI_UCX_recv_cmpl_cb(void *request, ucs_status_t
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_UCX_RECV_CMPL_CB);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_UCX_RECV_CMPL_CB);
 
-    if (ucp_request->req)
+    if (ucp_request->req) {
         rreq = ucp_request->req;
-    else
+    } else {
         rreq = MPL_malloc(sizeof(MPIR_Request), MPL_MEM_OTHER);
+        rreq->status.MPI_ERROR = MPI_SUCCESS;
+        MPIR_STATUS_SET_CANCEL_BIT(rreq->status, FALSE);
```

Comment:
could we just move this line down to line 44 in the original code?
```
MPIR_STATUS_SET_CANCEL_BIT(rreq->status, FALSE);
```

Reply:
We probably could although I think having it properly initialized at allocation time is cleaner and inline with what we do with `MPIR_Request_create`. If we move this line to line 44, then it means in the first branch (where `status == UCS_ERR_CANCELLED`) the error by be set to something erroneous, and in second branch (`MPI_ERR_TRUNCATE`) the cancel bits may be set. Thus some application may be misled by it.